package.json

```diff
@@ -122,6 +122,7 @@
     "perf-repeat": "yarn && yarn build && cross-env NODE_ENV=production node ./dist/bin-prettier.js --debug-repeat ${PERF_REPEAT:-1000} --loglevel debug ${PERF_FILE:-./index.js} > /dev/null",
     "perf-repeat-inspect": "yarn && yarn build && cross-env NODE_ENV=production node --inspect-brk ./dist/bin-prettier.js --debug-repeat ${PERF_REPEAT:-1000} --loglevel debug ${PERF_FILE:-./index.js} > /dev/null",
     "perf-benchmark": "yarn && yarn build && cross-env NODE_ENV=production node ./dist/bin-prettier.js --debug-benchmark --loglevel debug ${PERF_FILE:-./index.js} > /dev/null",
+    "checkjs": "tsc && tsc -p ./tsconfig-strict.json",
```

Comment:
"typecheck" would be more self-descriptive than "checkjs"

Reply:
Thanks for the quick feedback.

I would favor calling it "lint-types", would also maybe favor including it in the "lint" package script.

I would be interested if anyone else has feedback as well.

Reply:
We have `lint-docs` and `lint-dist`. `lint-type` wouldn't be consistent, it would look as if there were a folder or something called 'types'. Also typechecking is an established notion, separate from linting. I would understand 'linting types' as a check for issues in the JSDoc comments themselves, like what [`eslint-plugin-jsdoc`](https://www.npmjs.com/package/eslint-plugin-jsdoc) does.

Reply:
> including it in the "lint" package script

Better keep them separate. It'll look more clear on CI. And `yarn lint` is a little slow already, so I'd prefer to be able to run these checks independently. On the other hand, a command to run them all would be useful too.

Reply:
How about "check-types"?

Reply:
Better than "checkjs".

Reply:
I have now split it into check-types and check-types-strict. (I hope I didn't break the build, we will see.)

I have also updated check-types-strict (in a separate commit) to call check-types first, since check-types-strict would cover a very limited number of source files at this point.

I am now wondering if the check-types-strict should be moved into its own PR?

Reply:
Yep, I wouldn't touch `strict` until we have zero errors without it.

Reply:
The one thing is that strict mode is needed to ensure that I have added JSDoc comments for all functions in `src/doc/doc-builders.js`, like I proposed in this PR. If we drop strict mode, any ideas how we should verify proper proper documentation of `src/doc/doc-builders.js`?

Reply:
The typechecker will tell us when we work on other files and it finds a call to those functions with unexpected arguments.

Reply:
"how we should verify proper proper documentation of src/doc/doc-builders.js"

BTW that's not what typechecking does. It does not verify such things. It verifies that we use functions (classes, variables etc) the way we specified in their type annotations. Many type annotations are optional at that because the typechecker is able to infer many things, e.g. return types.

Reply:
Please disregard my previous comment. I misunderstood what you meant by 'documentation'. Obviously, you just meant 'adding types'.

Reply:
I am marking this conversation about the package scripts as resolved now.

I pushed a TBD comment to tsconfig.json to avoid forgetting to try type checking in strict mode in the fuutre.